### PR TITLE
Add codelab template

### PR DESCRIPTION
**Pull request by y3l2n**
added codelab template + updated message

### CODE REVIEW

File: src/viz/index.ts [hunk 1]
```diff
@@ -29,7 +29,9 @@ export const createFromTemplate = async (
   config.devBucket = addBucketPrefix(config.devBucket);
   config.prodBucket = addBucketPrefix(config.prodBucket);
   const {devBucket, prodBucket, projectName, basePath} = config;
-  const templatePath = path.join(basePath, 'templates', config.projectChoice);
+  const templatePath = config.codelab
```
Comment (diminishedprime) [inline src/viz/index.ts]: Snice the part of this that's different is just the template name, I would have that be defined through the ternary, not the whole thing. 

```
path.join(basePath, 'templates', newVariableFromTernary)
```

File: src/types.ts [hunk 2]
```diff
@@ -41,6 +41,7 @@ export interface ConnectorConfig
 }
 
 export interface VizConfig extends CommonConfig, VizConfigHasDefaults {
+  codelab: boolean;
```
Comment (diminishedprime) [inline src/types.ts]: This should probably be in the `VizConfigHasDefaults` type (and the default would be false)

File: templates/viz-codelab/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Config     | `src/index.json`    | [config reference]
+JavaScript | `src/index.js`      | [write viz code]
+CSS        | `src/index.css`     | [write css code]
+
+## Using this template
+
+To create a new community visualization based on this template, run the command
+
+```bash
+npx @google/dscc-gen viz
```
Comment (diminishedprime) [inline templates/viz-codelab/README.md]: This should include the `--codelab` flag.

File: templates/viz-codelab/src/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    var oldDiv = document.querySelector('div');
+    oldDiv.parentNode.removeChild(oldDiv);
+  }
+
+  // append the data to a div
+  let div = document.createElement('div');
+  div.innerHTML = `<h2>Community Visualization Codelab</h2>
+  This is the starter template for the <code>dscc-gen</code> codelab.
+  <p>
+  There are currently ${data.tables.DEFAULT.length} rows in the data.
```
Comment (diminishedprime) [inline templates/viz-codelab/src/index.js]: s/in the data/of data

File: templates/viz-codelab/src/manifest.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "dscc-gen-codelab",
+  "organization": "MyOrg",
+  "description": "My first visualization package.",
+  "logoUrl": "https://logo",
```
Comment (diminishedprime) [inline templates/viz-codelab/src/manifest.json]: Can you make this a static link to a graph icon or something? Once we have a gallery, it'd be nice if the codelab didn't give them something with broken images.

File: test/viz/index.spec.ts [hunk 6]
```diff
@@ -29,6 +29,7 @@ describe('End-to-end-tests for viz', () => {
       devBucket: 'test/dscc-gen-test-dev',
       prodBucket: 'test/dscc-gen-test-prod',
       yarn: false,
+      codelab: false,
```
Comment (diminishedprime) [inline test/viz/index.spec.ts]: You should also add a test for codelab: true and make sure it uses the right template. Probably by searching for a string/filename that only exists in one.

File: templates/viz/src/localMessage.js [hunk 7]
```diff
@@ -32,20 +32,135 @@ export const message = {
   fields: {
     dimID: [
       {
-        id: 'qt_nzqx6a0xvb',
+        id: 'qt_odsbu27hzb',
```
Comment (diminishedprime) [inline templates/viz/src/localMessage.js]: Can you do this change in a separate PR, it doesn't have anything to do with the codelab, it looks like.

File: test/viz/index.spec.ts [hunk 8]
```diff
@@ -10,6 +10,10 @@ console.log = jest.fn();
 const fiveMinutes = 5 * 60 * 1000;
 jest.setTimeout(fiveMinutes);
 
+const hasFile = async (...paths: string[]): Promise<boolean> => {
```
Comment (diminishedprime) [inline test/viz/index.spec.ts]: I think this functionality is already present in the `src/files.ts` file.

File: test/viz/index.spec.ts [hunk 9]
```diff
@@ -19,8 +23,41 @@ describe('End-to-end-tests for viz', () => {
...
+
+    await sut.createFromTemplate(config);
+    expect(await hasFile(vizName)).toBeTruthy();
+    expect(await hasFile(vizName, 'src', 'index.js')).toBeTruthy();
+
+    await files.remove(vizName);
+  });
+});
+
+describe('End-to-end-tests for viz codelab', () => {
```
Comment (diminishedprime) [inline test/viz/index.spec.ts]: Very minor nit, but this should probably be in the `end-to-end-tests for viz describe block, not its own describe block.

File: test/viz/index.spec.ts (hunk 8 above)
Comment (y3l2n) [reply test/viz/index.spec.ts]: hmm okay. This was just moving around code already in the viz test.
